LBH.AdultSocialCare.Api/V1/Gateways/DayCarePackageGateways/DayCarePackageGateway.cs

```diff
@@ -27,6 +30,22 @@ public async Task<Guid> CreateDayCarePackage(Infrastructure.Entities.DayCarePack
             return entry.Entity.DayCarePackageId;
         }
 
+        public async Task<DayCarePackageDomain> UpdateDayCarePackage(DayCarePackageForUpdateDomain dayCarePackageForUpdate)
+        {
+            var dayCarePackageEntity = await _dbContext.DayCarePackages
+                .Where(dc => dc.DayCarePackageId.Equals(dayCarePackageForUpdate.DayCarePackageId))
+                .SingleOrDefaultAsync().ConfigureAwait(false);
+            if (dayCarePackageEntity == null)
+            {
+                throw new ResourceNotFoundException($"Unable to locate day care package {dayCarePackageForUpdate.DayCarePackageId.ToString()}");
+            }
+            // Map fields with auto mapper and save
+            _mapper.Map(dayCarePackageForUpdate, dayCarePackageEntity);
+            dayCarePackageEntity.DateUpdated = DateTimeOffset.Now;
+            await _dbContext.SaveChangesAsync().ConfigureAwait(false);
```

Comment:
It may be good to determine success at this point (SaveChangesAsync should return 1 as 1 record has been updated), then we can have a turnery operator like so

```
bool success = await _dbContext.SaveChangesAsync().ConfigureAwait(false) == 1;
return success 
    ? dayCarePackageEntity.ToDomain() 
    : null;
```

Where null being returned indicates to the caller that there was a failure to save